meaning/meaning-gui.py

```diff
@@ -0,0 +1,136 @@
+import flet as ft
+import requests
+import bs4
+from bs4 import BeautifulSoup
+
+
+def get_meaning(word, site, datasrc):
+    """Get the meaning for a given word from an online dictionary."""
+    response = requests.get(site + word)
+    if not response.ok:
+        if response.status_code == 404:
+            print("Word not found")
+        else:
+            print(f"An error occurred: {response.reason}")
+        
+        return f"An error occurred: {response.reason}"
+
+    soup = BeautifulSoup(response.text, "html.parser")
+
+    data = soup.find("section", attrs=datasrc).findAll("div")
+
+    meaning = ""
+    for line in data:
+        meaning = f"{meaning}\n{line.get_text()}"
+
+    return meaning
+
+
+# Main
+def main(page):
+    page.title = "Meaning"
+    page.window_width=500
+    page.window_height=500
+    page.scroll = "adaptive"
+
+    # establish defaults in English
+    page.session.set("intro", "Your word's definition:\n")
```

Comment:
And TIL sessions in Flet, nice, thanks

Reply:
btw my next Flet app will be a wrapper around https://github.com/bbelderbos/movie-streaming/blob/main/script.py - to see what movie gets aired where ...